src/pages/ItemList.js

```diff
@@ -15,11 +15,14 @@ const ONE_MINUTE = ONE_SECOND * 60;
 const ONE_HOUR = ONE_MINUTE * 60;
 const ONE_DAY = ONE_HOUR * 24;
 
-
 function ItemList({ token }) {
   const [docs, setDocs] = useState([]);
+  const [filteredList, setFilteredList] = useState([]);
+  const [searchInput, setSearchInput] = useState('');
   const [now, setNow] = useState(Date.now());
   const [isLoading, setIsLoading] = useState(false);
+  let list = [];
+  filteredList ? (list = filteredList) : (list = docs);
```

Comment:
I wonder if we need to hold `filteredList` as state. Generally, it's a good idea to try to minimize the amount of state have. This is because state updates result in the re-rendering of the entire component and all of its children and the application is usually more performant with less state. 

Holding `now` as state was kind of a weird one and a hacky solution for another problem, but in a perfect work we wouldn't have that state either.

A lot of the time, I ask myself, "can I derive this new piece of state from other state?" If the answer is yes, then you likely don't need the new piece of state.

In this case, we can derive `filteredList` from `docs` and `searchInput`, so we could really just call `filterList` directly where we have `list.map` and have it return the appropriate list of items.

Reply:
Thanks for the feedback, got it.